### PR TITLE
Use modern Python 3.10 syntax

### PR DESCRIPTION
**Pull request by JonathanPlasse**
### Purpose of the pull request

This pull-request modernize the syntax used in the generated pyi files.

### Description about the pull request

- [x] PEP 585
- [x] PEP 604
- [ ] ~PEP 695~

### CODE REVIEW

**Comment by JonathanPlasse** [comment]
@nutti, this is ready for review.

File: src/fake_bpy_module/generator/writers.py [hunk 1]
```diff
@@ -137,14 +137,14 @@ def _write_function_code(self, func_node: FunctionNode):
             if not dtype_list_node.empty():
                 dtype_nodes = find_children(dtype_list_node, DataTypeNode)
                 if len(dtype_nodes) >= 2:
-                    dtype_str = f"typing.Union[{', '.join([n.to_string() for n in dtype_nodes])}]"
+                    dtype_str = ' | '.join(n.to_string() for n in dtype_nodes)
                 else:
                     dtype_str = dtype_nodes[0].to_string()
```
Comment (nutti) [inline src/fake_bpy_module/generator/writers.py:142]:
```
                if len(dtype_nodes) >= 2:
                    dtype_str = ' | '.join(n.to_string() for n in dtype_nodes)
                else:
                    dtype_str = dtype_nodes[0].to_string()
```

I think above code can be changed to a simpler code.
Same for others.

```
                dtype_str = ' | '.join(n.to_string() for n in dtype_nodes)
```

Comment (JonathanPlasse) [reply src/fake_bpy_module/generator/writers.py:142]: Indeed

**Comment by nutti** [comment]
@JonathanPlasse 

Thanks. The code looks good to me.
Should we change the document and pyproject file as well?

https://github.com/nutti/fake-bpy-module/blob/afb8a1c826d7afe1c1fff2d0da6d8bb13e180ae5/README.md?plain=1#L21
https://github.com/nutti/fake-bpy-module/blob/afb8a1c826d7afe1c1fff2d0da6d8bb13e180ae5/pyproject.toml#L7

**Comment by JonathanPlasse** [comment]
@nutti, no need, the modern syntax used in pyi files is never executed during runtime and is understood by MyPy and Pyright even with Python 3.8.

**Comment by nutti** [comment]
@JonathanPlasse 

OK, thank you.
BTW, does .pyi not depend on the Python version?
I'm not sure whether the Python version restriction can be removed or not.

**Comment by nutti** [comment]
@JonathanPlasse 

Thank you.
Maybe we should remove this restriction in future.
But I think this is not the scope of this issue.

**Review by nutti — APPROVED** [review]

**Pull request merged**